src/main/scala/firrtl/backends/smt/FirrtlToTransitionSystem.scala

```diff
@@ -0,0 +1,254 @@
+// See LICENSE for license details.
+// Author: Kevin Laeufer <laeufer@cs.berkeley.edu>
+
+package firrtl.backends.smt
+
+import firrtl.annotations.{Annotation, NoTargetAnnotation}
+import firrtl.options.Dependency
+import firrtl.stage.Forms
+import firrtl.{CircuitState, DependencyAPIMigration, Transform, ir}
+
+import scala.collection.mutable
+
+// Contains code to convert a flat firrtl module into a functional transition system which
+// can then be exported as SMTLib or Btor2 file.
+
+case class State(sym: SMTSymbol, init: Option[SMTExpr], next: Option[SMTExpr])
+case class Signal(name: String, e: BVExpr) { def toSymbol: BVSymbol = BVSymbol(name, e.width) }
+case class TransitionSystem(
+   name: String, inputs: Seq[BVSymbol], states: Seq[State], signals: Seq[Signal],
+   outputs: Set[String], assumes: Set[String], asserts: Set[String], fair: Set[String],
+   comments: Map[String, String] = Map())
+
+case class TransitionSystemAnnotation(sys: TransitionSystem) extends NoTargetAnnotation
+
+object FirrtlToTransitionSystem extends Transform with DependencyAPIMigration {
+  // TODO: We only really need [[Forms.MidForm]] + LowerTypes, but we also want to fail if there are CombLoops
```

Comment:
Is possible to make SMT compatible to `High Form`? I guess with this ability, we can use Z3 to prove the correctness between transforms.

Reply:
Working on `HighForm` would complicate this pass. However it would be possible to eventually add `HighFirrtl` support, just not in the first iteration.